### PR TITLE
Add `xcschemes` generator

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: tools/generators/xcschemes/src/Generator/TargetArgsAndEnv.swift
Comment (brentleyjones) [inline tools/generators/xcschemes/src/Generator/TargetArgsAndEnv.swift:7]: I'll do this FIXME before merging.

File: tools/generators/xcschemes/src/Generator/TransitivePreviewReferences.swift
Comment (brentleyjones) [inline tools/generators/xcschemes/src/Generator/TransitivePreviewReferences.swift:5]: I'll do this FIXME before merging.

File: tools/generators/xcschemes/src/Generator/CreateScheme.swift
Comment (brentleyjones) [inline tools/generators/xcschemes/src/Generator/CreateScheme.swift:115]: I'll do this FIXME before merging.

**Review by brentleyjones — COMMENTED** [review]
@MobileNativeFoundation/rules_xcodeproj-maintainers I have a couple FIXME to address, and lots of tests to add, but this is ready to be reviewed.

File: tools/generators/xcschemes/README.md
Comment (thiagohmcruz) [inline tools/generators/xcschemes/README.md:3]:
```suggestion
The `xcschemes` generator generates `.xcscheme` files for a project.
```

File: tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift
Comment (thiagohmcruz) [inline tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift:267]: What would be the advantage or doing that instead of treating `None` and `[]` as the same "empty" input?

**Review by thiagohmcruz — APPROVED** [review]

File: tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift
Comment (brentleyjones) [reply tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift:267]: Allows saying in the scheme that you don't want any arguments, regardless if the rule would provide some. Right now that's not possible, both `None` and `[]` would mean no custom ones, but you will get whatever the rule provides.

**Comment by brentleyjones** [comment]
https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2686 shows the Starlark side of it.

File: tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift
Comment (brentleyjones) [reply tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift:267]: I plan to have a special `inherit` value, which will also be the default. So if you set a falsey value, it won't inherit.

Comment (thiagohmcruz) [reply tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift:267]: SGTM, thx for clarifying!

Comment (brentleyjones) [reply tools/generators/xcschemes/src/Generator/CreateCustomSchemeInfos.swift]: This is now in. `nil` means "inherit", and non-`nil` means that was explicitly specified by the user, so use those values. On the Starlark side `inherit` eventually becomes `nil` (because of a `-1` count), and `None`/`[]` both mean "don't inherit any args, I mean literally have no args for this action).

**Review by luispadron — APPROVED** [review]

**Comment by brentleyjones** [comment]
https://github.com/MobileNativeFoundation/rules_xcodeproj/issues/2814

**Pull request merged**
